tests/log/Log.cpp

```diff
@@ -71,9 +111,41 @@ TEST_CASE("Testing the Log<>() function", "[api][log]") {
     NUClear::PowerPlant plant(config);
 
     // We are installing with an initial log level of debug
-    plant.install<TestReactor, NUClear::DEBUG>();
-
-    plant.emit(std::make_unique<int>(5));
-
+    plant.install<TestReactor>();
     plant.start();
+
+    // Test that we have the correct number of messages
+    REQUIRE(messages.size() == 78);
+
+
+    // Test that each of the messages are correct for each log level
+    int i = 0;
+    for (const auto& display_level : levels) {
+        // Test logs from reactions directly
+        for (const auto& log_level : levels) {
+            if (display_level <= log_level) {
```

Comment:
Is it not somehow possible for the logs to log with the incorrect log level and for you to now pick it up?

Reply:
It would screw with the order of the underlying array and blow it up